addons/mail/static/tests/owl/component/activity_tests.js

```diff
@@ -0,0 +1,832 @@
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_detailsAssignationUserAvatar').length,
+        1,
+        "activity details should have user avatar"
+    );
+});
+
+QUnit.test('activity with mail template layout', async function (assert) {
+    assert.expect(8);
+
+    await this.start();
+    const activityLocalId = this.env.store.dispatch('_createActivity', {
+        mail_template_ids: [{ id: 1, name: "Dummy mail template" }]
+    });
+    await this.createActivityComponent(activityLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_Activity').length,
+        1,
+        "should have activity component in DOM"
+    );
+    const activity = document.querySelector('.o_Activity');
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_sidebar').length,
+        1,
+        "should have activity sidebar in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_mailTemplates').length,
+        1,
+        "should have activity mail templates in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_Activity_mailTemplate').length,
+        1,
+        "should have activity mail template in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_MailTemplate_name').length,
+        1,
+        "should have activity mail template name in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelector('.o_MailTemplate_name').textContent,
+        "Dummy mail template",
+        "should have activity mail template name in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_MailTemplate_preview').length,
+        1,
+        "should have activity mail template name preview button in DOM"
+    );
+    assert.strictEqual(
+        activity.querySelectorAll('.o_MailTemplate_send').length,
+        1,
+        "should have activity mail template name send button in DOM"
+    );
+
+
+});
+
+QUnit.test('activity with mail template: preview mail', async function (assert) {
+    assert.expect(3);
+
+    await this.start({
+        intercepts: {
+            do_action: function (event) {
+                assert.deepEqual(
+                    event.data.action,
+                    {
+                        context: {
+                            default_res_id: 42,
+                            default_model: 'res.partner',
+                            default_use_template: true,
+                            default_template_id: 1,
+                            force_email: true,
+                        },
+                        name: 'Compose Email',
+                        type: 'ir.actions.act_window',
+                        res_model: 'mail.compose.message',
+                        views: [[false, 'form']],
+                        target: 'new',
+                    },
+                    "should open the preview mail form view"
+                );
```

Comment:
not really fond of `deepEqual`.
For instance, changing `force_email` to `false` later on necessarily makes this test fail... Why?
It doesn't make sense in my opinion, because the purpose of this test is to check whether it opens the full-composer on activity, whose behaviour is not perceived as changed this example.

Reply:
Just checking the generated action is as it should be.
How would you replace that ?

Reply:
assertions on action `type`, `res_model`, `context.default_res_id`, `context.default_model`, `context.default_use_template`, `context.default_template_id` make sense. The rest are details in my opinion.

Also I would not assume that this test single intercepted `do_action` is just dedicated for opening of mail template form view, even if that's the case for the moment. The correct way is to detect an action dedicated for this and make an `assert.step()` with related `assert.verifySteps()`.